spec/datadog/appsec/contrib/active_record/mysql2_adapter_spec.rb

```diff
@@ -119,15 +119,33 @@
       User.find_by_sql("SELECT * FROM users WHERE name = 'Bob'").to_a
     end
 
-    it 'adds an event to processor context if waf result is a match' do
-      result = Datadog::AppSec::SecurityEngine::Result::Match.new(
-        events: [], actions: {}, derivatives: {}, timeout: false, duration_ns: 0, duration_ext_ns: 0
-      )
+    context 'when waf result is a match' do
+      let(:result) do
+        Datadog::AppSec::SecurityEngine::Result::Match.new(
+          events: [],
+          actions: { 'generate_stack' => { 'stack_id' => 'some-id' } },
+          derivatives: {},
+          timeout: false,
+          duration_ns: 0,
+          duration_ext_ns: 0
+        )
+      end
 
-      expect(Datadog::AppSec.active_context).to receive(:run_rasp).and_return(result)
-      expect(Datadog::AppSec.active_context.events).to receive(:<<).and_call_original
+      before do
+        allow(Datadog::AppSec.active_context).to receive(:run_rasp).and_return(result)
+      end
 
-      User.where(name: 'Bob').to_a
+      it 'adds an event to processor context if waf result is a match' do
```

Comment:
I think we have a context from the `context` and shrink the explanation. Also the wording is slightly outdated

```suggestion
      it 'adds an event to the context events' do
```